### PR TITLE
rm support for debian 10 buster

### PR DESCRIPTION
**Pull request by rrotter**
- update puppet metadata.json
- remove vm config template
- remove buster specific code branches

### CODE REVIEW

**Review by skorner — COMMENTED** [review]
Thumbs-up from me on sunsetting buster support in our environment. I'll defer to someone else for a technical review of the changes and leave one question.

File: metadata.json [hunk 1]
```diff
@@ -36,7 +36,7 @@
   "operatingsystem_support": [
     {
       "operatingsystem": "Debian",
-      "operatingsystemrelease": ["11","12"]
```
Comment (skorner) [inline metadata.json]: Why drop 11?

File: spec/defines/virtual_machine_spec.rb [hunk 2]
```diff
@@ -298,11 +298,7 @@ def contain_preseed
       context "with pre-bullseye build name" do
         let(:params) { {build: "buster"} }
 
-        [
-          %r{\sin-target systemctl enable puppet\b}
-        ].each do |line|
-          it { is_expected.not_to contain_preseed.with_content(line) }
-        end
+        it { is_expected.not_to compile }
```
Comment (daaang) [inline spec/defines/virtual_machine_spec.rb:301]: :D

**Review by daaang — APPROVED** [review]

File: metadata.json (hunk 1 above)
Comment (rrotter) [reply metadata.json]: Looks like I blindly removed the first array element. Fixed, thanks.

**Pull request merged**